Enemy ai

khoile3009 opened the pull request:
Finish the first enemy, put one into the demo level.
 - Enemy behavior contain direction, stamina, and state controlling the enemy.
 - DeerAggro, Patrol, Investigate is each controller for each state on enemy.
 - TargetDetection is to detect player if walk into vision range.
 - TargetLock is to lock on player

Commit:
adding enemy ai branch from other repository

khoile3009 committed:
Finish detection and chasing. Working on smoothing the patrolling behavior

khoile3009 committed:
Enemy behavior

khoile3009 committed:
metafile

khoile3009 committed:
Scene file

khoile3009 committed:
Merge branch 'master' into enemy-ai

khoile3009 committed:
Refactor detection so that it detect fire cracker as well

khoile3009 committed:
Remove Old player detector

khoile3009 committed:
Refactor enemy script for easier development

khoile3009 committed:
Add placeholder sandwich

khoile3009 committed:
Put enemy inside demo level

khoile3009 committed:
Add sprite controller

khoile3009 committed:
Merge branch 'master' into enemy-ai

khoile3009 committed:
Fix merge conflict and put enemy into demo 1 for example

VirulentKid commented:
One thing I noticed was that when you're near the enemy but no route available, the enemy will just convulse

khoile3009 committed:
Merge branch 'master' into enemy-ai

khoile3009 committed:
Turn circle collider into front box collider. Comment to code. Move my assets into main assets

khoile3009 committed:
Merge branch 'master' into enemy-ai

khoile3009 commented:
I just update it and commented. Feel free to take a look

lucaspopp0 reviewed (dismissed):
Approving because this looks good so far, but I have a few comments:
1. Can you make enemies do damage to the player?
2. The enemy gets stuck if you put a wall in front of it
3. It looks like the enemy basically pushes the sandwich instead of like eating it or something. Can we make them be triggered by berries instead, and slowly eat the berry?

lucaspopp0 committed:
Update .gitignore

lucaspopp0 reviewed (approved):
Approving because this looks good so far, but I have a few comments:
1. Can you make enemies do damage to the player?
2. The enemy gets stuck if you put a wall in front of it
3. It looks like the enemy basically pushes the sandwich instead of like eating it or something. Can we make them be triggered by berries instead, and slowly eat the berry?
